### PR TITLE
Split compilation and tests into separate phases

### LINKED ISSUE #742: Add a switch to continue running universe even when tests fail
And have a report of failed tests when finished.

/cc @natemcmaster @ryanbrandenburg

### PR DESCRIPTION
**Pull request by natemcmaster**
Resolves #742, prerequisite for fixing MusicStore #773 and Template tests #713 

Before:

```
Common: build
Common: test
DI: build
DI: test
FileSystem: build
FileSystem: test   <--- fails and build.cmd halts
```

After:
```
Common: build
DI: build
FileSystem: build
Kestrel: build
...etc..

Common: test
DI: test
FileSystem: test <--- fails
Kestrel: test <-- tests should continue chugging
...etc...
```

Also, enables running: `build.cmd /t:test /p:NoBuild=true` to skip re-building altogether and to jump ahead to tests.

### CODE REVIEW

File: build/RepositoryBuild.targets [hunk 1]
```diff
@@ -84,4 +94,26 @@
...
 
+  <Target Name="_TestRepository" DependsOnTargets="GetRepoBuildProps">
+    <PropertyGroup>
+      <BuildArguments>/t:Test /p:NoBuild=$(_NoBuildRepos) $(RepositoryBuildArguments)</BuildArguments>
+
+      <_BuildScriptToExecute Condition="'$(OS)'!='Windows_NT'">build.sh</_BuildScriptToExecute>
+      <_BuildScriptToExecute Condition="'$(OS)'=='Windows_NT'">build.cmd</_BuildScriptToExecute>
+    </PropertyGroup>
+
+    <Message Text="============ Testing $(RepositoryToBuild) ============" Importance="High" />
```
Comment (pakrym) [inline build/RepositoryBuild.targets]: Do we have to copy? Doesn't Korebuild have an argument which repo to build?

Comment (pakrym) [reply build/RepositoryBuild.targets]: + Seems to be duplicated

Comment (natemcmaster) [reply build/RepositoryBuild.targets]: No, KoreBuild does not have an argument that allows you to build one repo. It would if we did #739

Comment (natemcmaster) [reply build/RepositoryBuild.targets]: Updated to reduce some duplication between these targets

**Comment by pakrym** [comment]
LGTM

**Pull request merged**